libraries/abstractions/transport/include/transport_interface.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef TRANSPORT_INTERFACE_H_
+#define TRANSPORT_INTERFACE_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+
+
+/**
+ * @brief Maximum length of an ASCII DNS name.
+ */
+#define MAX_DNS_NAME_LENGTH    ( 253 )
+
+/**
+ * @brief The NetworkContext is an incomplete type. An implementation of this
+ * interface must define what pContext points to as per the requirements. This context
+ * is passed into the network interface functions.
+ */
+typedef struct NetworkContext
+{
+    /**
+     * @brief Points to the context for each interface.
+     */
+    void * pContext;
+} NetworkContext_t;
```

Comment:
The `transport_interface.h` file in C-SDK uses the following forward declaration of `NetworkContext_t`
```
struct NetworkContext;
typedef struct NetworkContext NetworkContext_t;
```
Could you update the latest version of C-SDK as the transport interface file?